src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java

```diff
@@ -32,27 +32,51 @@
 
 /**
  * Head corresponding to a pull request.
- * Named like {@code PR-123}.
+ * Named like {@code PR-123} or {@code PR-123-merged} or {@code PR-123-unmerged}.
  */
 public final class PullRequestSCMHead extends SCMHead {
 
-    private static final String PR_BRANCH_PREFIX = "PR-";
-
     private static final long serialVersionUID = 1;
 
     private final PullRequestAction metadata;
+    private Boolean merge;
+    private final boolean trusted;
 
-    PullRequestSCMHead(GHPullRequest pr) {
-        super(PR_BRANCH_PREFIX + pr.getNumber());
+    PullRequestSCMHead(GHPullRequest pr, String name, boolean merge, boolean trusted) {
+        super(name);
         metadata = new PullRequestAction(pr);
+        this.merge = merge;
+        this.trusted = trusted;
     }
 
     public int getNumber() {
         if (metadata != null) {
             return Integer.parseInt(metadata.getId());
         } else { // settings compatibility
-            return Integer.parseInt(getName().substring(PR_BRANCH_PREFIX.length()));
+            return Integer.parseInt(getName().substring("PR-".length()));
```

Comment:
Very old settings predating `PullRequestAction` must have use the `/PR-\d+/` scheme.